applications/luci-app-openvpn/luasrc/view/openvpn/cbi-select-input-add.htm

```diff
@@ -1,11 +1,102 @@
-<div class="cbi-section-create">
-	<% if self.invalid_cts then -%><div class="cbi-section-error"><% end %>
-	<input type="text" class="cbi-section-create-name" name="cbi.cts.<%=self.config%>.<%=self.sectiontype%>.text" />
-	<select class="cbi-section-create-name" name="cbi.cts.<%=self.config%>.<%=self.sectiontype%>.select">
-	<%- for k, v in luci.util.kspairs(self.add_select_options) do %>
-		<option value="<%=k%>"><%=luci.util.pcdata(v)%></option>
-	<% end -%>
-	</select>
-	<input class="cbi-button cbi-button-add" type="submit" value="<%:Add%>" title="<%:Add%>" />
-	<% if self.invalid_cts then %><br /><%:Invalid%></div><% end %>
+<script type="text/javascript">
+//<![CDATA[
+	function vpn_add()
+	{
+		var vpn_name     = div_add.querySelector("#instance_name1").value.replace(/[^\x00-\x7F]|[\s!@#$%^&*()+=\[\]{};':"\\|,<>\/?]/g,'');
+		var vpn_template = div_add.querySelector("#instance_template").value;
+		var form         = document.getElementsByName('cbi')[0];
+
+		if (!vpn_name || !vpn_name.length)
+		{
+			return info_message(vpn_output, "The 'Name' field must not be empty!", 2000);
+		}
+
+		document.getElementById("instance_name1").value = vpn_name;
+		if (document.getElementById("cbi-openvpn-" + vpn_name) != null)
+		{
+			return info_message(vpn_output, "Instance with that name already exists!", 2000);
+		}
+
+		if (!vpn_template || !vpn_template.length)
+		{
+			return info_message(vpn_output, "Please select a valid VPN template!", 2000);
+		}
+
+		if (form)
+		{
+			form.submit();
+		}
+	}
+
+	function vpn_upload()
+	{
+		var vpn_name = div_upload.querySelector("#instance_name2").value.replace(/[^\x00-\x7F]|[\s!@#$%^&*()+=\[\]{};':"\\|,<>\/?]/g,'');
+		var form     = document.getElementsByName('cbi')[0];
+
+		if (!vpn_name || !vpn_name.length)
+		{
+			return info_message(vpn_output, "The 'Name' field must not be empty!", 2000);
+		}
+
+		document.getElementById("instance_name2").value = vpn_name;
+		if (document.getElementById("cbi-openvpn-" + vpn_name) != null)
+		{
+			return info_message(vpn_output, "Instance with that name already exists!", 2000);
+		}
+
+		if (form)
+		{
+			form.enctype = 'multipart/form-data';
+			form.action  = '<%=url('admin/services/openvpn/upload')%>';
+			form.submit();
+		}
+	}
+
+	function info_message(output, msg, timeout)
+	{
+		timeout = timeout || 0;
+		output.innerHTML = '<em>' + msg + '</em>';
+		if (timeout > 0)
+		{
+			setTimeout(function(){ output.innerHTML=""}, timeout);
+		}
+	}
+//]]>
+</script>
+
+<div class="cbi-section-node">
+	<div class="table cbi-section-table">
+		<h4 style="white-space: nowrap; border-bottom: 0px; margin: 10px 5px 5px 5px;"><%:Template based configuration%></h4>
```

Comment:
Can you move these inline styles into `editor_css.htm` ?

Reply:
I've separated all styles in a css section within cbi-select-input-add.htm. In principle it's no problem to shift this into an external css, but I have no idea how to reference that file.